openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/EntityTimeSeriesRepository.java

```diff
@@ -58,34 +60,47 @@ public EntityTimeSeriesRepository(
     this.entityType = entityType;
     this.searchRepository = Entity.getSearchRepository();
     this.daoCollection = Entity.getCollectionDAO();
+    this.allowedFields = getEntityFields(entityClass);
     Entity.registerEntity(entityClass, entityType, this);
   }
 
   @Transaction
-  public T createNewRecord(T recordEntity, String extension, String recordFQN) {
-    recordEntity.setId(UUID.randomUUID());
-    storeInternal(recordEntity, recordFQN, extension);
-    storeRelationshipInternal(recordEntity);
-    postCreate(recordEntity);
-    return recordEntity;
+  public T createNewRecord(T recordEntity, String recordFQN) {
+    return createNewRecord(recordEntity, recordFQN, null);
   }
 
-  public T createNewRecord(T recordEntity, String recordFQN) {
+  @Transaction
+  public T createNewRecord(T recordEntity, String recordFQN, String extension) {
     recordEntity.setId(UUID.randomUUID());
-    storeInternal(recordEntity, recordFQN);
+    if (extension != null) {
```

Comment:
any reason to have this if condition, given storeInternal also checking the extension is null or not

Reply:
I can remove it and pass `null` with the other function calling `createNewRecord` -- it is mainly convenience to not have to update the calling function. But we can remove that overhead

Reply:
Ah I see -- my bad. Makes sense